### ISSUE TITLE
DS3622xs+ installation fails, probably corrupted.

### ISSUE THREAD

**Issue by amadino13** [pocopico/tinycore-redpill#364]
**Describe the bug**
Installation fails. Failed to install DSM. The File is probably corrupted. 

**To Reproduce**
Followed the regular installation steps. Tried 7.2. When that failed, I tried 7.1.1. Same result. I tried with downloading the firmware from Synology website and even just using the auto download. I am using JOT because when I use withfriends my system freezes. 

**Screenshots**
[Screenshot 2023-07-29 075910](https://github.com/pocopico/tinycore-redpill/assets/64305457/ba50f30f-09ef-47f7-9f6b-6dabeffa1f97)

**Additional context**
using i7 13700k, 64gb ram, LSI 9208-8i card, Intelx520 NIC, 10x HGST HUH721008ALE600

**Comment by damienmillet** [comment]
hi, did you download patched version of install dsm from withfriends ?

**Comment by shadowspassion89** [comment]
How do you get a patched version?

**Comment by pocopico** [comment]
Did you try the html builder ?

**Comment by shadowspassion89** [comment]
I built the image using the instructions on the first page ./rploader.sh build <then the version>. Then I tired the tinycore friend boot option. But I get the same issue as amadino13. I tried the .pat in /mnt/sda3 and a downloaded.pat. The recovery says they are corrupted.

**Comment by shadowspassion89** [comment]
This the vmdk image btw, imported into Proxmox.

tc@box:~$ ./rploader.sh identifyusb
Machine is VIRTUAL Hypervisor=KVM
Sorry, no usb disk could be identified

SynologyNAS> fdisk -l
Disk /dev/sdk: 64 GB, 68719476736 bytes, 134217728 sectors
8354 cylinders, 255 heads, 63 sectors/track
Units: sectors of 1 * 512 = 512 bytes

Device  Boot StartCHS    EndCHS        StartLBA     EndLBA    Sectors  Size Id Type
/dev/sdk1    0,130,3     1023,254,63       8192   16785407   16777216 8192M fd Linux raid autodetect
/dev/sdk2    1023,254,63 1023,254,63   16785408   20979711    4194304 2048M fd Linux raid autodetect
Disk /dev/sdf: 1024 MB, 1073741824 bytes, 2097152 sectors
1011 cylinders, 34 heads, 61 sectors/track
Units: sectors of 1 * 512 = 512 bytes

Device  Boot StartCHS    EndCHS        StartLBA     EndLBA    Sectors  Size Id Type
/dev/sdf1 *  0,32,33     6,62,56           2048     100351      98304 48.0M 83 Linux
Partition 1 has different physical/logical start (non-Linux?):
     phys=(0,32,33) logical=(0,33,36)
Partition 1 has different physical/logical end:
     phys=(6,62,56) logical=(48,13,7)
/dev/sdf2    6,62,57     15,205,62       100352     253951     153600 75.0M 83 Linux
Partition 2 has different physical/logical start (non-Linux?):
     phys=(6,62,57) logical=(48,13,8)
Partition 2 has different physical/logical end:
     phys=(15,205,62) logical=(122,15,9)
/dev/sdf3    15,205,63   130,138,8       253952    2097151    1843200  900M 83 Linux
Partition 3 has different physical/logical start (non-Linux?):
     phys=(15,205,63) logical=(122,15,10)
Partition 3 has different physical/logical end:
     phys=(130,138,8) logical=(1011,5,33)
Disk /dev/md0: 8191 MB, 8589869056 bytes, 16777088 sectors
2097136 cylinders, 2 heads, 4 sectors/track
Units: sectors of 1 * 512 = 512 bytes

Disk /dev/md0 doesn't contain a valid partition table
Disk /dev/md1: 2047 MB, 2147418112 bytes, 4194176 sectors
524272 cylinders, 2 heads, 4 sectors/track
Units: sectors of 1 * 512 = 512 bytes

Disk /dev/md1 doesn't contain a valid partition table

**Comment by shadowspassion89** [comment]
Changing the boot device from, SCSI 0 to SATA 0 in Proxmox seemed to do the trick for the VMDK.

**Comment by pocopico** [comment]
Did you follow any guide for proxmox ? 

Here are some notes I’ve kept
 
 # set vm id
id=930

# create image directory, download and uncomporess
mkdir -p /var/lib/vz/images/${id}
curl --location https://github.com/pocopico/tinycore-redpill/raw/main/tinycore-redpill.v0.4.4.img.gz --output /var/lib/vz/images/${id}/tinycore-redpill.img.gz
gzip --decompress /var/lib/vz/images/${id}/tinycore-redpill.img.gz --keep

# create vm
qm create ${id} \
  --args "-device 'qemu-xhci,addr=0x18' -drive 'id=synoboot,file=/var/lib/vz/images/${id}/tinycore-redpill.img,if=none,format=raw' -device 'usb-storage,drive=synoboot,bootindex=1'" \
  --cores 2 \
  --cpu host \
  --machine q35 \
  --memory 2048 \
  --name DSM \
  --net0 virtio,bridge=vmbr0 \
  --numa 0 \
  --onboot 0 \
  --ostype l26 \
  --scsihw virtio-scsi-pci \
  --sata0 local-lvm:vm-${id}-disk-0,discard=on,size=100G,ssd=1 \
  --sockets 1 \
  --serial0 socket \
  --serial1 socket \
  --tablet 1

# create disk for sata0
pvesm alloc local-lvm ${id} vm-${id}-disk-0 100G

**Comment by MrHappy** [comment]
I tried booting from sata0 and scsi0, tried installing using the image downloaded manually, tried the patched pat (copied from the vm), tried the by DSM downloaded pat. All result (on a different point) with a message that the image is corrupted. Config of my vm is like @pocopico gave above, with the exception the boot disk (redpill-image) is on sata0/scsi0.

**Comment by lzzsalvo** [comment]
Try downloading DSM directly from Synology Download Center. https://www.synology.com/support/download 

Sometimes it happens that the automatic download gives the corrupted file error.

**Comment by MrHappy** [comment]
I 'used' 3 different images.

1. Downloaded manually from Synology
2. scp'd the pat-file from the vm 
3. Have DSM download it automatically

All three options failed with the corrupted file error (all be it on a different stage/percentage).

**Comment by MrHappy** [comment]
After reading [this post](https://xpenology.com/forum/topic/62871-tinycore-redpill-loader-tcrp-development-release-09/?do=findComment&comment=448744) I tried Peter Suh's version (again) and now it installs fine. No 'corruption' errors.

**Comment by zfdang** [comment]
> 

I'm having the same corrupted problem. How to apply this patch?

**Comment by zfdang** [comment]
> After reading [this post](https://xpenology.com/forum/topic/62871-tinycore-redpill-loader-tcrp-development-release-09/?do=findComment&comment=448744) I tried Peter Suh's version (again) and now it installs fine. No 'corruption' errors.

@pocopico is it possible to include this patch into next release?

**Comment by maluueu** [comment]
> > @pocopico Is there any update on this? Currently, one cannot build DS3622xs+
> 
> I just installed 7.2 successfully using image of PeterSuh:
> 
> https://github.com/PeterSuh-Q3/tinycore-redpill

Is there a way to change from pocopico's TCRP to PeterSuh's TCRP on the fly?

I'm currently using pocopico's TCRP in a Proxmox VM with just the image mounted as SATA0 HDD and a SATA1 HDD for DSM, no USB.

Can I just switch the SATA0 HDD Image to PeterSuh's TCRP?

**Comment by zfdang** [comment]
> > > @pocopico Is there any update on this? Currently, one cannot build DS3622xs+
> > 
> > 
> > I just installed 7.2 successfully using image of PeterSuh:
> > https://github.com/PeterSuh-Q3/tinycore-redpill
> 
> Is there a way to change from pocopico's TCRP to PeterSuh's TCRP on the fly?
> 
> I'm currently using pocopico's TCRP in a Proxmox VM with just the image mounted as SATA0 HDD and a SATA1 HDD for DSM, no USB.
> 
> Can I just switch the SATA0 HDD Image to PeterSuh's TCRP?

I just reflashed PeterSuh's image to replace the original TCRP image in USB disk. I guess you can also replace the mounted image to have a try.

**Comment by maluueu** [comment]
> > > > @pocopico Is there any update on this? Currently, one cannot build DS3622xs+
> > > 
> > > 
> > > I just installed 7.2 successfully using image of PeterSuh:
> > > https://github.com/PeterSuh-Q3/tinycore-redpill
> > 
> > 
> > Is there a way to change from pocopico's TCRP to PeterSuh's TCRP on the fly?
> > I'm currently using pocopico's TCRP in a Proxmox VM with just the image mounted as SATA0 HDD and a SATA1 HDD for DSM, no USB.
> > Can I just switch the SATA0 HDD Image to PeterSuh's TCRP?
> 
> I just reflashed PeterSuh's image to replace the original TCRP image in USB disk. I guess you can also replace the mounted image to have a try.

Yes, just did it. Works perfectly fine, just important the .img as disk with qm Importdisk and that's it.
Did you get ACPI to work? Or are you not using Proxmox
